modules/core/02-client/keeper/proposal.go

```diff
@@ -98,5 +98,11 @@ func (k Keeper) HandleUpgradeProposal(ctx sdk.Context, p *types.UpgradeProposal)
 
 	// sets the new upgraded client in last height committed on this chain is at plan.Height,
 	// since the chain will panic at plan.Height and new chain will resume at plan.Height
-	return k.upgradeKeeper.SetUpgradedClient(ctx, p.Plan.Height, bz)
+	err = k.upgradeKeeper.SetUpgradedClient(ctx, p.Plan.Height, bz)
+	if err == nil {
+		// emitting an event for handling client upgrade proposal
+		EmitUpgradeClientProposalEvent(ctx, p.Title, p.Plan.Height)
+	}
+
+	return err
```

Comment:
I think this would be more "idiomatic" go to return early with the err and then we could emit the event before returning nil. 

Something like below:

```suggestion
	if err := k.upgradeKeeper.SetUpgradedClient(ctx, p.Plan.Height, bz); err != nil {
		return err
	}

        // emitting an event for handling client upgrade proposal
	EmitUpgradeClientProposalEvent(ctx, p.Title, p.Plan.Height)
	
	return nil
```